staging/src/k8s.io/apimachinery/pkg/fields/selector.go

```diff
@@ -68,6 +72,13 @@ func (t *hasTerm) Matches(ls Fields) bool {
 	return ls.Get(t.field) == t.value
 }
 
+func (t *hasTerm) Supported(ls Fields) error {
+	if !ls.Has(t.field) {
```

Comment:
This assumes the full set of values are populated in the field set regardless of value. I thought we already had validation of whether a particular selector was supported via the field selector conversion mechanisms. Granted, those aren't great, but I'd rather see those improved or replaced than add a second different mechanism like this on top